libraries/node-core-library/src/Async.ts

```diff
@@ -158,6 +193,68 @@ export class Async {
     });
   }
 
+  /**
+   * Given an input array and a `callback` function, invoke the callback to start a
+   * promise for each element in the array.
+   *
+   * @remarks
+   * This API is similar to the system `Array#forEach`, except that the loop is asynchronous,
+   * and the maximum number of concurrent promises can be throttled
+   * using {@link IAsyncParallelismOptions.concurrency}.
+   *
+   * If `callback` throws a synchronous exception, or if it returns a promise that rejects,
+   * then the loop stops immediately.  Any remaining array items will be skipped, and
+   * overall operation will reject with the first error that was encountered.
+   *
+   * @param iterable - the array of inputs for the callback function
+   * @param callback - a function that starts an asynchronous promise for an element
+   *   from the array
+   * @param options - options for customizing the control flow
+   */
+  public static async forEachAsync<TEntry>(
+    iterable: Iterable<TEntry> | AsyncIterable<TEntry>,
+    callback: (entry: TEntry, arrayIndex: number) => Promise<void>,
+    options?: IAsyncParallelismOptions
```

Comment:
```suggestion
    options?: IAsyncParallelismOptions & { weighted?: false }
```

Should fix the type validation.

That change also exposes that `mapAsync` should probably also have a similar overload (with a corresponding changelog entry).

Reply:
I added a changelog entry for `mapAsync`

I've wrestled with the types a bit and not getting anywhere, the overload narrowing is probably blocked by this https://github.com/microsoft/TypeScript/issues/22609 which is causing issues when calling `forEachAsync` from `mapAsync`. I added a `ts-expect-error`, that's not a great long term solution so open to other ideas!